python/ray/dataframe/dataframe.py

```diff
@@ -690,17 +691,80 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
                     limit_direction='forward', downcast=None, **kwargs):
         raise NotImplementedError("Not Yet implemented.")
 
+    def iterrows(self):
+        """Iterate over DataFrame rows as (index, Series) pairs.
+
+        NOTE:
+            Generators can't be pickeled so from the remote function
+            we expand the generator into a list before getting it.
+            This is not that ideal.
+
+        Returns:
+            A generator that iterates over the rows of the frame.
+        """
+        iters = ray.get([
+            _deploy_func.remote(
+                lambda df: list(df.iterrows()), part) for part in self._df])
+        return itertools.chain.from_iterable(iters)
+
     def items(self):
-        raise NotImplementedError("Not Yet implemented.")
+        """Iterator over (column name, Series) pairs.
+
+        NOTE:
+            Generators can't be pickeled so from the remote function
+            we expand the generator into a list before getting it.
+            This is not that ideal.
+
+        Returns:
+            A generator that iterates over the columns of the frame.
+        """
+        iters = ray.get([_deploy_func.remote(
+            lambda df: list(df.items()), part) for part in self._df])
+
+        def concat_iters(iterables):
+            for partitions in zip(*iterables):
+                to_concat = []
+                for items in partitions:
+                    index, _series = items
+                    to_concat.append(_series)
+                series = pd.concat(to_concat)
+                yield (index, series)
+
+        return concat_iters(iters)
 
     def iteritems(self):
-        raise NotImplementedError("Not Yet implemented.")
+        """Iterator over (column name, Series) pairs.
 
-    def iterrows(self):
-        raise NotImplementedError("Not Yet implemented.")
+        NOTE:
+            Returns the same thing as .items()
+
+        Returns:
+            A generator that iterates over the columns of the frame.
+        """
+        return self.items()
 
     def itertuples(self, index=True, name='Pandas'):
-        raise NotImplementedError("Not Yet implemented.")
+        """Iterate over DataFrame rows as namedtuples, with index value as first
+        element of the tuple.
```

Comment:
Can we fit this in one line? We copied the style from http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html.

Reply:
I separated this line because flake8 gets upset when I leave `Iterate over DataFrame rows as namedtuples, with index value as first element of the tuple.` as one line: `(94 > 79 characters)`, should I just ignore flake8?

Reply:
I meant change the text to fit in one line.